From 71a6aa9349d5846115fa09bdf4506328802ffa7f Mon Sep 17 00:00:00 2001
From: Panu Matilainen <pmatilai@redhat.com>
Date: Wed, 30 Oct 2024 09:10:14 +0200
Subject: [PATCH] Silence extra output on rpmspec query on Buildsystem specs

Dynamic spec parsing emits "Reading <path>" messages to make it clear
something out of the ordinary is going on, but this is undesireable
on rpmspec queries. Luckily there's a trivial solution: rpmbuild where
you want these messages runs in verbose mode by default whereas rpmspec
does not. So just moving the message to INFO level does exactly the
right thing in both the above cases. Add a test to go.

Fies: #3413
---
 build/parseSpec.c | 2 +-
 tests/rpmbuild.at | 9 ++++++++-
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/build/parseSpec.c b/build/parseSpec.c
index df08575ec..0d4605366 100644
--- a/build/parseSpec.c
+++ b/build/parseSpec.c
@@ -1426,7 +1426,7 @@ static rpmRC parseSpecParts(rpmSpec spec, const char *pattern,
     /* rpmGlob returns files sorted */
     if (rpmGlob(pattern, &argc, &argv) == 0) {
 	for (int i = 0; i < argc; i++) {
-	    rpmlog(RPMLOG_NOTICE, "Reading %s\n", argv[i]);
+	    rpmlog(RPMLOG_INFO, "Reading %s\n", argv[i]);
 	    pushOFI(spec, argv[i]);
 	    snprintf(spec->fileStack->readBuf, spec->fileStack->readBufLen,
 		     "# Spec part read from %s\n\n", argv[i]);
diff --git a/tests/rpmbuild.at b/tests/rpmbuild.at
index 6c86e6eb5..3fa90bb9b 100644
--- a/tests/rpmbuild.at
+++ b/tests/rpmbuild.at
@@ -78,7 +78,7 @@ runroot rpm -q --qf "%{packager}\n" /build/SRPMS/hello-1.0-1.src.rpm
 RPMTEST_CLEANUP
 
 AT_SETUP([rpmbuild -b buildsystem])
-AT_KEYWORDS([build])
+AT_KEYWORDS([build buildsystem])
 RPMDB_INIT
 
 cp "${RPMTEST}/data/macros.buildsystem" "${RPMTEST}/${RPM_CONFIGDIR_PATH}/macros.d/"
@@ -185,6 +185,13 @@ runroot rpmbuild -bb \
 [error: line 11: Required parametric macro %buildsystem_autotools_build not defined for buildsystem autotools
 ])
 
+RPMTEST_CHECK([
+runroot rpmspec -q --qf "%{nvr}\n" /data/SPECS/amhello.spec
+],
+[0],
+[amhello-1.0-1
+],
+[])
 RPMTEST_CLEANUP
 
 AT_SETUP([rpmbuild -b steps])
-- 
2.47.1

